tools/azsdk-cli/docs/specs/6-package-updater.spec.md

```diff
@@ -0,0 +1,689 @@
+
+| Language   | Limitation | Impact | Workaround |
+|------------|------------|--------|------------|
+| JavaScript | Code formatting isn't hooked up to the emitter | Noisy diffs or linter failures | Run code formatting step in the `update-metadata.ps1` script |
+
+---
+
+## Design Proposal
+
+### Overview
+
+Implement four individual CLI + MCP tools. Each tool:
+
+1. Resolves package path & language.
+2. Loads the configuration for the language.
+3. Chooses inline command over script path if both present.
+4. Executes the relevant sub-script(s) in an ordered, fault-isolated manner.
+5. Aggregates structured result JSON with a `result`, human-readable `message`, and optional `next_steps` hint guiding subsequent tool invocation.
+
+Provides a singular CLI + MCP command that orchestrates the four tools (update-ci → update-changelog → update-version → update-metadata) and returns a combined JSON summary.
+
+### Detailed Design
+
+The four tools are intentionally small, composable units. Each emits a machine-readable JSON payload to enable deterministic chaining by an agent or CI workflow. All tools share common behaviors:
+
+| Area | Expected Behavior |
+|--------|-------------------|
+| Invocation Mode | CLI and MCP (agent) alias; same semantics |
+| Input Path Validation | Must exist and contain a recognizable SDK language structure; errors early if not |
+| Config Resolution | Load `swagger_to_sdk_config.json`; prefer `inline command` over `script path` when both are configured |
+| Timeout (default) | 5 minutes per tool invocation |
+| Output Schema | JSON with fields: result, message (see schema below) |
+| Idempotency | Multiple successive runs produce identical filesystem state except for timestamps/logs |
+| Script Implementation | MUST be PowerShell `.ps1`. SHOULD accept named parameters (e.g., `sdkRepoPath`). MUST use exit code `0` for success/noop; non-zero only on outright failure. Legacy non-PS scripts should be wrapped by a small PowerShell shim. Scripts MUST write a concise stderr line prefixed with `ERROR:`; include suggested mitigation steps and links to docs in the message where possible. |
+| Script Execution Environment | All scripts are executed using PowerShell Core (`pwsh`) terminal |
+| Next Steps Hint | Plain language phrase embedded in `message` (e.g., `next_steps: update version`) |
+| Missing Required Tools | If a required external tool is not present (for example: `dotnet`, `gofmt`, `pwsh`), prompt the user to run the `verify-setup` tool which validates and documents missing prerequisites. |
+
+#### 1. Changelog Update Tool
+
+Name (CLI): `azsdk package update-changelog-content`
+
+Name (MCP): `azsdk_package_update_changelog`
+
+Purpose: Ensure changelog is auto-generated and has an entry for the upcoming release (management-plane) or provide guidance (data-plane) where manual editing is expected. This tool only modifies changelog content and does not change release date or version numbers.
```

Comment:
this is the ultimate goal, but it would be better if the spec right now reflects what needs to be implemented for `Scenario 1` in which case, the purpose would be different, but each language will be able to know what the expectation is of what they need to implement

Reply:
From mgmt side work on update changelog, We already have update changelog logics implemented in current generation script in CI. (except dotnet /cc @haiyuazhang). So our work is to split existing scripts, extract the logics, write it using powershell, and test. 

The API on how language script integrates with MCP can be found here: https://github.com/Azure/azure-sdk-tools/blob/f152f2e8df4208c3ab7fc211b4620d891365a748/tools/azsdk-cli/docs/specs/6-package-updater.spec.md#update-changelog-script

e.g. I'm working on the PR to update changelog script. https://github.com/Azure/azure-sdk-for-java/pull/47079/files

Reply:
> this is the ultimate goal

This is the goal of Scenario 1. For mgmt-plane SDKs, auto-generation is the goal; while for data-plane, the goal is to provide guidance.